pkg/reconciler/instances/ory/db/db_test.go

```diff
@@ -0,0 +1,176 @@
+package db
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	postgresYaml = `
+        global:
+          postgresql:
+            postgresqlDatabase: db4hydra
+            postgresqlUsername: hydra
+            postgresqlPassword: secretpw
+          ory:
+            hydra:
+              persistence:
+                enabled: true
+                postgresql:
+                  enabled: true`
+
+	gcloudYaml = `
+        global:
+          ory:
+            hydra:
+              persistence:
+                enabled: true
+                gcloud:
+                  enabled: true
+                  saJson: testsa
+                user: hydra
+                password: secretpw
+                dbUrl: ory-gcloud-sqlproxy.kyma-system:5432
+                dbName: db4hydra
+                dbType: postgres`
+
+	mysqlDBYaml = `
+        global:
+          ory:
+            hydra:
+              persistence:
+                enabled: true
+                gcloud:
+                  enabled: false
+                user: hydra
+                password: secretpw
+                dbUrl: mydb.mynamespace:1234
+                dbName: db4hydra
+                dbType: mysql`
+
+	customDBYaml = `
+        global:
+          ory:
+            hydra:
+              persistence:
+                enabled: true
+                gcloud:
+                  enabled: false
+                user: hydra
+                password: secretpw
+                dbUrl: mydb.mynamespace:1234
+                dbName: db4hydra
+                dbType: cockroach`
+
+	noDBYaml = `
+        global:
+          ory:
+            hydra:
+              persistence:
+                enabled: false`
+)
+
+func TestDBSecret(t *testing.T) {
+	t.Run("Deployment with Postgres SQL in cluster", func(t *testing.T) {
+		// given
+		t.Parallel()
+		name := types.NamespacedName{Name: "test-postgres-secret", Namespace: "test"}
+		var values map[string]interface{}
+		err := yaml.Unmarshal([]byte(postgresYaml), &values)
+		require.NoError(t, err)
+
+		// when
+		cfg, errNew := new(values)
+		dsnExpected := cfg.preparePostgresDSN()
+		secret, errGet := Get(name, values)
```

Comment:
`Get` function does not really resemble what it does under the hood. If it implemented some method from the interface that has proper docs it would be okay but here we have a standalone exported function. Is it possible to rename it to make it more meaningful?